app/assets/javascripts/shopify_app/redirect.js

```diff
@@ -12,15 +14,10 @@
       // If the current window is the 'parent', change the URL by setting location.href
       window.top.location.href = targetInfo.url;
     } else {
-      // If the current window is the 'child', change the parent's URL with postMessage
-      normalizedLink = document.createElement('a');
-      normalizedLink.href = targetInfo.url;
-
-      data = JSON.stringify({
-        message: 'Shopify.API.remoteRedirect',
-        data: {location: normalizedLink.href}
-      });
-      window.parent.postMessage(data, targetInfo.myshopifyUrl);
+      // If the current window is the 'child' or embedded, change the parent's URL with App Bridge redirect
+      // This case can happen when an app updates its access scopes, or when the shop thinks the app is installed, but
+      // the app does not have shop entry
+      window.appBridgeRedirect(targetInfo.url);
```

Comment:
Now that I understand [the "link trick"](https://github.com/Shopify/shopify_app/pull/1247/files#r789210446), I'm wondering if this line has a bug 🤔  ...

Can `targetInfo.url` be relative?
If so, the old code would have prepended the current host to it, and the new code doesn't.